tests/core.test.js

```diff
@@ -0,0 +1,183 @@
+const mockParsedParams = { id: "mockParsedParams" };
+const mockRegion = "mock region";
+const mockQuery = "mock query";
+
+beforeEach(() => {
+  core = rewire("../core.js");
+})
+
+describe("mapToAwsMethod", () =>{
+  const mockMethodPromise = jest.fn();
+
+  const mockClient = {
+    mockMethod: jest.fn().mockImplementation((payload) => ({
+      promise: mockMethodPromise
+    })),
+    anotherMockMethod: jest.fn().mockImplementation(() => ({
+      promise: () => {}
+    })),
+  };
+
+  const simpleParameters = {
+    name: "some name"
+  };
+
+  const parametersWithoutEmptyValues = {
+    string: "Some text",
+    number: 5,
+    bool: true,
+    array: ["Some value", "Some other value"],
+    nestedObject: {
+      key: "value"
+    }
+  };
+
+  test("returns a function that triggers only specified service method on call with provided parameters", () => {
+
+    const returnedFunction1 = core.mapToAwsMethod("mockMethod");
+    const returnedFunction2 = core.mapToAwsMethod("anotherMockMethod");
+
+    returnedFunction1(mockClient, parametersWithoutEmptyValues);
+
+    expect(mockClient.mockMethod).toHaveBeenCalledTimes(1);
+    expect(mockClient.mockMethod).toHaveBeenCalledWith(parametersWithoutEmptyValues);
+    expect(mockClient.anotherMockMethod).toHaveBeenCalledTimes(0);
+
+    returnedFunction2(mockClient, simpleParameters);
+
+    expect(mockClient.mockMethod).toHaveBeenCalledTimes(1);
+    expect(mockClient.anotherMockMethod).toHaveBeenCalledTimes(1);
+    expect(mockClient.anotherMockMethod).toHaveBeenCalledWith(simpleParameters);
+  });
+
+  test("returns a function that triggers the payload function with params on call and pass its result to service method", () => {
+    const mockPayloadFunction = jest.fn().mockImplementation((params) => parametersWithoutEmptyValues);
+    const returnedFunction = core.mapToAwsMethod("mockMethod", mockPayloadFunction);
+    const someRegion = "some region";
+
+    expect(mockPayloadFunction).toHaveBeenCalledTimes(0);
+
+    returnedFunction(mockClient, simpleParameters, someRegion);
+    expect(mockPayloadFunction).toHaveBeenCalledTimes(1);
+    expect(mockPayloadFunction).toHaveBeenCalledWith(simpleParameters, someRegion);
+    expect(mockClient.mockMethod).toHaveBeenCalledWith(parametersWithoutEmptyValues);
+  });
+
+  test("returns a function that calls `promise()` method on service function result on call", () => {
+    const returnedFunction = core.mapToAwsMethod("mockMethod");
+
+    returnedFunction(mockClient);
+    expect(mockClient.mockMethod).toHaveBeenCalled();
+    expect(mockMethodPromise).toHaveBeenCalled();
+  });
+
+  test("returns a function that throws if no such method has been found", () => {
+    const returnedFunction = core.mapToAwsMethod("nonExistentMethod");
+    expect(() => returnedFunction(mockClient)).toThrowError();
+  });
+});
+
+describe("bootstrap", () => {
+
+  test("returns a single object containing all of the functions passed as arguments", () => {
```

Comment:
It actually does more - `...functions with injected client passed...`, but that's going to be hard to understand.
What about `Injects empty object as client parameter into pluginMethods and autocompleteFunctions`?

Reply:
You are right that this function does, but it's not what this specific test tests.